### ISSUE TITLE
[BUG] "Oberved a panic" on Upgrade to 2.10.0

### ISSUE THREAD

**Issue by strowi** [rancher/rancher#48180]
**Rancher Server Setup**
- Rancher version: 2.9.3/2.10.0
- Installation option (Docker install/Helm Chart): Helm Chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): v1.30.4-k3s1

**Information about the Cluster**
- Kubernetes version: v1.30.4-k3s1
- Cluster Type (Local/Downstream): local

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**
Upgrading Rancher from 2.9.3 to 2.10.0 fails with the new pod crashing continuously.

**To Reproduce**

- deploy rancher 2.9.3 on 1.30.4-k3s1:
```yaml
apiVersion: helm.cattle.io/v1
kind: HelmChart
metadata:
  name: rancher
  namespace: kube-system
spec:
  chart: rancher
  repo: https://releases.rancher.com/server-charts/stable
  targetNamespace: cattle-system
  valuesContent: |-
    test: 1
    hostname: "rancher.xyz.tld"
    replicas: 1
    bootstrapPassword: "xyz"
    ingress:
      tls:
        source: "letsEncrypt"
    letsEncrypt:
      email: "xyz"
    auditLog:
      level: 1
  version: 2.9.3
```
- upgrade to 2.10.0
- check new rancher-pod crashlooping

**Result**
New Rancher-Pod is in a CrashLoop.

**Expected Result**

Rancher being upgraded and running.

**Screenshots**

**Additional context**

```
Defaulted container "rancher" out of: rancher, rancher-audit-log
2024/11/26 18:48:17 [INFO] Rancher version v2.10.0 (df45e368c82d4027410fa4700371982b9236b7c8) is starting
2024/11/26 18:48:17 [INFO] Rancher arguments {ACMEDomains:[] AddLocal:true Embedded:false BindHost: HTTPListenPort:80 HTTPSListenPort:443 K8sMode:auto Debug:false Trace:false NoCACerts:true AuditLogPath:/var/log/auditlog/rancher-api-audit.log AuditLogMaxage:1 AuditLogMaxsize:100 AuditLogMaxbackup:1 AuditLevel:1 Features: ClusterRegistry:}
2024/11/26 18:48:17 [INFO] Listening on /tmp/log.sock
2024/11/26 18:48:17 [INFO] Running in clustered mode with ID 10.42.0.189, monitoring endpoint cattle-system/rancher
2024/11/26 18:48:19 [INFO] Applying CRD features.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD clusterrepos.catalog.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD clusterroletemplatebindings.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD globalroles.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD globalrolebindings.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD projects.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD projectroletemplatebindings.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD roletemplates.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD clusterproxyconfigs.management.cattle.io
2024/11/26 18:48:21 [INFO] Updating embedded CRD uiplugins.catalog.cattle.io
2024/11/26 18:48:23 [INFO] Applying CRD navlinks.ui.cattle.io
2024/11/26 18:48:23 [INFO] Applying CRD podsecurityadmissionconfigurationtemplates.management.cattle.io
2024/11/26 18:48:24 [INFO] Applying CRD clusters.management.cattle.io
2024/11/26 18:48:24 [INFO] Applying CRD apiservices.management.cattle.io
2024/11/26 18:48:24 [INFO] Applying CRD clusterregistrationtokens.management.cattle.io
2024/11/26 18:48:25 [INFO] Applying CRD settings.management.cattle.io
2024/11/26 18:48:25 [INFO] Applying CRD preferences.management.cattle.io
2024/11/26 18:48:25 [INFO] Applying CRD features.management.cattle.io
2024/11/26 18:48:25 [INFO] Applying CRD operations.catalog.cattle.io
2024/11/26 18:48:26 [INFO] Applying CRD apps.catalog.cattle.io
2024/11/26 18:48:26 [INFO] Applying CRD fleetworkspaces.management.cattle.io
2024/11/26 18:48:26 [INFO] Applying CRD managedcharts.management.cattle.io
2024/11/26 18:48:27 [INFO] Applying CRD clusters.provisioning.cattle.io
2024/11/26 18:48:27 [INFO] Applying CRD clusters.provisioning.cattle.io
2024/11/26 18:48:27 [INFO] Applying CRD rkeclusters.rke.cattle.io
2024/11/26 18:48:28 [INFO] Applying CRD rkecontrolplanes.rke.cattle.io
2024/11/26 18:48:28 [INFO] Applying CRD rkebootstraps.rke.cattle.io
2024/11/26 18:48:28 [INFO] Applying CRD rkebootstraptemplates.rke.cattle.io
2024/11/26 18:48:29 [INFO] Applying CRD rkecontrolplanes.rke.cattle.io
2024/11/26 18:48:29 [INFO] Applying CRD custommachines.rke.cattle.io
2024/11/26 18:48:30 [INFO] Applying CRD etcdsnapshots.rke.cattle.io
E1126 18:48:41.875925      39 panic.go:128] "Observed a panic" panic="runtime error: slice bounds out of range [:6] with length 5" panicGoValue="runtime.boundsError{x:6, y:5, signed:true, code:0x1}" stacktrace=<
  goroutine 855 [running]:
  k8s.io/apimachinery/pkg/util/runtime.logPanic({0xb4ef548, 0x10c85620}, {0x9c06e00, 0xc00493c9d8})
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:107 +0xbc
  k8s.io/apimachinery/pkg/util/runtime.handleCrash({0xb4ef548, 0x10c85620}, {0x9c06e00, 0xc00493c9d8}, {0x10c85620, 0x0, 0x10000c004930b88?})
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:82 +0x5e
  k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc000f7b3f0?})
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/runtime/runtime.go:59 +0x108
  panic({0x9c06e00?, 0xc00493c9d8?})
    /usr/lib64/go/1.22/src/runtime/panic.go:770 +0x132
  github.com/rancher/rancher/pkg/auth/providers/local.userSearchIndexer.func1({0xc004721218?, 0xc002f23678?}, 0x6)
    /app/pkg/auth/providers/local/local_auth_provider.go:415 +0x145
  github.com/rancher/rancher/pkg/auth/providers/local.userSearchIndexer({0xa1d4b40?, 0xc00b9189c0?})
    /app/pkg/auth/providers/local/local_auth_provider.go:422 +0x2d5
  k8s.io/client-go/tools/cache.(*storeIndex).updateSingleIndex(0xc001e92440, {0xa40166f, 0x2c}, {0x0, 0x0}, {0xa1d4b40, 0xc00b9189c0}, {0xc004722e90, 0xc})
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/thread_safe_store.go:162 +0xfe
  k8s.io/client-go/tools/cache.(*storeIndex).updateIndices(...)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/thread_safe_store.go:196
  k8s.io/client-go/tools/cache.(*threadSafeMap).Update(0xc005531080, {0xc004722e90, 0xc}, {0xa1d4b40, 0xc00b9189c0})
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/thread_safe_store.go:241 +0x1a5
  k8s.io/client-go/tools/cache.(*threadSafeMap).Add(0xa1d4b40?, {0xc004722e90?, 0xa1d4b40?}, {0xa1d4b40?, 0xc00b9189c0?})
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/thread_safe_store.go:233 +0x25
  k8s.io/client-go/tools/cache.(*cache).Add(0xc0063beb58, {0xa1d4b40, 0xc00b9189c0})
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/store.go:174 +0xb1
  k8s.io/client-go/tools/cache.processDeltas({0xb4d75d0, 0xc000f55080}, {0xb50bf60, 0xc0063beb58}, {0xc001275ea0?, 0xc002ae6e00?, 0xc000f7b3f0?}, 0x1)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/controller.go:560 +0x1b2
  k8s.io/client-go/tools/cache.(*sharedIndexInformer).HandleDeltas(0xc000f55080, {0x88e6e60?, 0xc004930b70?}, 0x1)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/shared_informer.go:643 +0x1b1
  k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc000f7b3f0, 0xc00bbfbae0)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/delta_fifo.go:616 +0x524
  k8s.io/client-go/tools/cache.(*controller).processLoop(0xc000f7b550)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/controller.go:195 +0x30
  k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30?)
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:226 +0x33
  k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc002f23e40, {0xb46ede0, 0xc0019ab7d0}, 0x1, 0xc0000c4300)
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:227 +0xaf
  k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc002be7e40, 0x3b9aca00, 0x0, 0x1, 0xc0000c4300)
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:204 +0x7f
  k8s.io/apimachinery/pkg/util/wait.Until(...)
    /go/pkg/mod/k8s.io/apimachinery@v0.31.1/pkg/util/wait/backoff.go:161
  k8s.io/client-go/tools/cache.(*controller).Run(0xc000f7b550, 0xc0000c4300)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/controller.go:166 +0x37e
  k8s.io/client-go/tools/cache.(*sharedIndexInformer).Run(0xc000f55080, 0xc0000c4300)
    /go/pkg/mod/k8s.io/client-go@v0.31.1/tools/cache/shared_informer.go:508 +0x2c8
  github.com/rancher/lasso/pkg/cache.(*deferredCache).Run(0xc0063beb70, 0xc0000c4300)
    /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20240924233157-8f384efc8813/pkg/cache/cache.go:145 +0x3c
  created by github.com/rancher/lasso/pkg/cache.(*sharedCacheFactory).Start in goroutine 1
    /go/pkg/mod/github.com/rancher/lasso@v0.0.0-20240924233157-8f384efc8813/pkg/cache/sharedinformerfactory.go:115 +0x230
 >
```

**Comment by bigkevmcd** [comment]
@strowi It's lower-casing names to make the searching better.

I'm guessing that you have non-ASCII Displaynames or Usernames for users?

I'm working on this just now, will try and replicate and get a fix out.

**Comment by strowi** [comment]
@bigkevmcd : Thanks, indeed, the hint with the usernames was on-point. I checked the users (which are coming fom AzureAD), not posting the full name, but one contained these letters:  	`İ çç`. 

After deleting the user via rancher-ui i tried the update again and it worked. Not sure though what happens if the user gets synced again. :)

**Comment by nickwsuse** [comment]
Verified on v2.10-head ID: `0d4f38a` ✅ 
----

Test Cases (All are upgrades from `v2.9.3 -> v.2.10-head` and were run on docker single node and HA clusters):
1. Basic Auth | Regular ASCII Characters ✅ 
2. Basic Auth | Non-ASCII Characters ✅ 
3. SAML | Regular ASCII Characters ✅ 
4. SAML | Non-ASCII Characters ✅ 
5. OAuth | Regular ASCII Characters ✅ 
6. OAuth | Non-ASCII Characters ✅ 

The PR only made changes to local auth from the looks of it, but I went ahead and tested SAML and OAuth just to be sure that other providers aren't affected.
